### PR TITLE
Allow a recompression if we have change/add/remove KeyEndEffectorsMatchNameArray.

### PR DESCRIPTION
**Pull request by Meradrin**
Currently we sadly need to change a parameter of the compression setting or edit the ForceRebuildVersion for recompress when we edit the KeyEndEffectorsMatchNameArray. So I simply add all bone name to the hash. It is a hash without case sensentive for the bone name.

### CODE REVIEW

File: ACLPlugin/Source/ACLPlugin/Private/AnimCompress_ACL.cpp [hunk 1]
```diff
@@ -184,7 +185,12 @@ void UAnimCompress_ACL::PopulateDDCKey(FArchive& Ar)
 	uint32 ForceRebuildVersion = 1;
 	uint16 AlgorithmVersion = get_algorithm_version(AlgorithmType8::UniformlySampled);
 	uint32 SettingsHash = Settings.get_hash();
-
+	
+	for (const FString& MatchName : UAnimationSettings::Get()->KeyEndEffectorsMatchNameArray)
```
Comment (nfrechette) [inline ACLPlugin/Source/ACLPlugin/Private/AnimCompress_ACL.cpp]: That is a great idea! Can you add a separate value for the hash to keep it apart and append it to the Ar archive on line 195?

**Comment by nfrechette** [comment]
This is a great idea and perhaps something we should consider adding upstream for all ue4 codecs, perhaps as part of the DDC key. I'll see with Epic if they are interested in that idea.

**Comment by Meradrin** [comment]
I think that will be a great idea to be add on compression codec. I don't think all compression codec take this value in consideration so maybe have a flag, bool or virtual function for telling that you are using this value for the compression and that need to be take in consideration with the DDC.

File: ACLPlugin/Source/ACLPlugin/Private/AnimCompress_ACL.cpp [hunk 2]
```diff
@@ -183,9 +184,14 @@ void UAnimCompress_ACL::PopulateDDCKey(FArchive& Ar)
 
 	uint32 ForceRebuildVersion = 1;
 	uint16 AlgorithmVersion = get_algorithm_version(AlgorithmType8::UniformlySampled);
-	uint32 SettingsHash = Settings.get_hash();
```
Comment (nfrechette) [inline ACLPlugin/Source/ACLPlugin/Private/AnimCompress_ACL.cpp]: Looks like the settings hash was removed by mistake

Comment (Meradrin) [reply ACLPlugin/Source/ACLPlugin/Private/AnimCompress_ACL.cpp]: Sorry :(

**Comment by nfrechette** [comment]
Great stuff, thanks!

**Comment by Meradrin** [comment]
Oky, I have fix it now, sadly I have a separate folder for ACL UE4 Pluging that the one I use in the engine and I copy past in notepad the modification in the github repo. My bad for the error!

**Pull request merged**